### PR TITLE
Modlogin 41 Handling: Failed login attempts - Lock Account

### PR DESCRIPTION
**Pull request by OleksiiKuzminov**

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/LoginAPI.java [hunk 1]
```diff
@@ -38,9 +40,12 @@
 import io.vertx.core.logging.LoggerFactory;
 import java.io.Reader;
 import java.net.URL;
+
 import java.util.MissingResourceException;
+
 import static org.folio.rest.RestVerticle.MODULE_SPECIFIC_ARGS;
 import org.folio.rest.jaxrs.model.Errors;
+import static org.folio.util.LoginAttemptsHelper.*;
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/rest/impl/LoginAPI.java]: Please avoid of this star imports

File: src/test/java/org/folio/logintest/RestVerticleTest.java [hunk 2]
```diff
@@ -36,33 +36,34 @@
 import static org.folio.logintest.UserMock.bombadilId;
 import static org.folio.logintest.UserMock.gollumId;
 import static org.folio.logintest.UserMock.sarumanId;
-import org.folio.rest.impl.LoginAPI;
+import static java.lang.Thread.sleep;
+import static org.folio.logintest.UserMock.*;
```
Comment (Igor-Gorchakov) [inline src/test/java/org/folio/logintest/RestVerticleTest.java]: The same

File: src/main/java/org/folio/util/LoginAttemptsHelper.java [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+  private static final String LOGIN_ATTEMPTS_USERID_FIELD = "'userId'";
+  private static final Logger logger = LoggerFactory.getLogger(LoginAttemptsHelper.class);
+
+  /**
+   * Method build criteria for lookup Login Attempts for user by user id
+   *
+   * @param userId - identifier of user
+   * @return - Criterion for search login attempts using user id
+   */
+  public static Criterion buildCriteriaForUserAttempts(String userId) throws Exception {
```
[the remark this answers is not in the document]
Comment (OleksiiKuzminov) [reply src/main/java/org/folio/util/LoginAttemptsHelper.java]: Criteria constructor throws Exception

File: src/main/java/org/folio/util/LoginAttemptsHelper.java [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+import java.util.UUID;
+
+import static org.folio.rest.RestVerticle.MODULE_SPECIFIC_ARGS;
+import static org.folio.rest.impl.LoginAPI.OKAPI_TENANT_HEADER;
+import static org.folio.rest.impl.LoginAPI.OKAPI_TOKEN_HEADER;
+
+/**
+ * Helper class that contains static methods which helps with processing Login Attempts business logic
+ */
+public class LoginAttemptsHelper {
```
[the remark this answers is not in the document]
Comment (OleksiiKuzminov) [reply src/main/java/org/folio/util/LoginAttemptsHelper.java]: no need private constructor for class only with static methods

**Comment by OleksiiKuzminov** [comment]
@kurtnordstrom *What happens if the call to mod-configuration fails? Does this block login?*
it try to get this settings from module config or get default value if both variants are null. So it isn't block login anyway

**Pull request merged**